aries_vcx/src/handlers/issuance/mediated_holder.rs

```diff
@@ -0,0 +1,99 @@
+use agency_client::agency_client::AgencyClient;
+use aries_vcx_core::anoncreds::base_anoncreds::BaseAnonCreds;
+use aries_vcx_core::ledger::base_ledger::AnoncredsLedgerRead;
+use aries_vcx_core::wallet::base_wallet::BaseWallet;
+use messages::msg_fields::protocols::cred_issuance::CredentialIssuance;
+use messages::msg_fields::protocols::notification::Notification;
+use messages::AriesMessage;
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use crate::errors::error::prelude::*;
+use crate::handlers::connection::mediated_connection::MediatedConnection;
+use crate::handlers::issuance::holder::Holder;
+use crate::handlers::util::{matches_opt_thread_id, matches_thread_id};
+use crate::protocols::issuance::holder::state_machine::HolderState;
+
+pub async fn holder_update_with_mediator(
```

Comment:
This function is only used in tests (and same for issuer, verifier, prover), so perhaps it should be placed near the test code (under `#[cfg(test)]`)?

Reply:
Also libvcx though, so not yet

https://github.com/hyperledger/aries-vcx/pull/945#discussion_r1302203423

Reply:
These `_update_with_mediator` are positively not used in libvcx

Reply:
Ahh yeah! You are right! I got it mixed up with `*_find_message_to_handle` which are.

So you are right, I'll put these functions under `#[cfg(test)]`

Reply:
Decent improvement indeed, applied